framework/doc/content/source/base/MooseApp.md

```diff
@@ -1,4 +1,4 @@
 # MooseApp
 
-The `MooseApp` object is the base class for !!acro MOOSE!! based applications and serves as the
+The `MooseApp` object is the base class for [!ac!MOOSE] based applications and serves as the
```

Comment:
Why is it `[!ac!MOOSE]` and not `[!ac](MOOSE)` analogous to the cite syntax?

Reply:
The main reasons is that I wanted to make the acronym syntax as compact as possible so it is quick to use. There are also some implementation reasons that are not super important. 

If you feel it makes the markdown better I will change it, but can we merge this first? I will then add support for both and print a deprecated warning for the old version.

Reply:
I added the new version, we should probably change it. The consistency is better then the one character savings. 

We can merge this with both and then go update the apps to avoid submodule stuff.